scylla-cql/src/frame/response/authenticate.rs

```diff
@@ -42,3 +44,96 @@ impl AuthChallenge {
         })
     }
 }
+
+pub type AuthError = String;
+
+/// Trait used to represent a user-defined custom authentication.
+/// evaluate_challenge: to handle an authentication challenge initiated by the server.
+/// The information contained in the token parameter is authentication protocol specific.
+/// It may be NULL or empty.
+///
+/// success: to handle the success phase of exchange.
+/// The token parameters contain information that may be used to finalize the request.
```

Comment:
Instead of putting descriptions of methods to the main docstring, you should place them beside `evaluate_challenge` and `success`. Please look at e.g. `prepared_statement.rs` for examples.

Reply:
Please make sure to apply this advice to other public types that you introduced.